docs/workflows/best-practices-architecture.md

```diff
@@ -0,0 +1,76 @@
+
+Tracking your architecture evolution is therefore key to ensuring your software landscapes and its systems remain future-proof. Every architecture is bound to change over time, due to a changing environment in which it operates. These guidelines are meant to help architects and development teams with their component architecture, based on best practices.
+
+<sig-toc></sig-toc>
+
+## Why modularize your codebase?
+
+Every non-trivial system is split into components. However, there are multiple different reasons for why people decide to modularize their system. Some of these reasons strictly relate to technical benefits, and other reasons purely relate to making life easier for the people working on the system. 
+
+- **Deploying components separately gives more runtime flexibility.** Runtime load is rarely spread equally across all components in the system. Deploying components separately allows for more flexibility in the deployment architecture, which is generally beneficial when it comes to managing performance behavior, scalability, and resilience. 
+- **Deploying components independently increases development flexibility.** If components are deployed separately, teams can work on those components independently and deploy 'their' component without having to simultaneously deploy the other components. However, it is quite rare for components within a system to operate in complete isolation from each other. Components tend to have various dependencies to other components. This can take the form of an interface dependency, but something like a shared database also acts as a dependency between components. The decision to deploy components separately tends to make those components a bit more independent, but not fully independent. 
+- **Dividing your codebase into logical "parts" makes it easier for people to locate functionality.** This is the most common reason to modularize systems and is done purely for the benefit of the people working on the system. Dividing a system into components helps to manage the cognitive load: People can first think about the system in terms of its high-level architecture, without being confronted with the full implementation details for every single part. The component structure is also helpful to communicate where certain functionality is implemented. 
+
+Many systems use a combination of these reasons when designing their component architecture. For example, it is quite common to define the top-level architecture as a number of microservices that are deployed separately. But every microservice also has an internal architecture, which is mainly based on dividing the codebase into logical areas. Such a combination is not "wrong". It is a perfectly reasonable choice which indicates that modularizing a system needs to consider both technical and human aspects.
```

Comment:
* "Many systems use a combination of these reasons when designing their component architecture." use of the word systems is a bit strange here: systems do not design their architecture, people do (pun not intended, no relation between architecture and guns implied...).  --> "The design of a component architecture involves a combination of these reasons."
* "mainly based on dividing the codebase into logical areas." --> to further clarify, perhaps "mainly based on dividing the codebase into logical areas _that are not deployed independently_."?